src/plone/restapi/deserializer/atcontent.py

```diff
@@ -1,175 +1,178 @@
-            accessor = field.getAccessor(self.context)
-
-        return accessor()
-
-    def __contains__(self, key):
-        return key in self.context.Schema()
-
-    def get(self, key, default=None):
-        try:
-            return self[key]
-        except KeyError:
-            return default
+
+                name = field.getName()
+
+                if name in data:
+                    deserializer = queryMultiAdapter((field, obj, self.request),
+                                                     IFieldDeserializer)
+                    if deserializer is None:
+                        continue
+                    value, kwargs = deserializer(data[name])
+                    mutator = field.getMutator(obj)
+                    mutator(value, **kwargs)
+                    modified = True
+
+            if modified:
+                errors = self.validate()
+                if not validate_all:
+                    errors = {f: e for f, e in errors.items() if f in data}
+                if errors:
+                    errors = [{
+                        'message': e,
+                        'field': f,
+                        'error': 'ValidationError'} for f, e in errors.items()]
+                    raise BadRequest(errors)
+
+                if create:
+                    if obj.checkCreationFlag():
+                        obj.unmarkCreationFlag()
+                    notify(ObjectInitializedEvent(obj))
+                    obj.at_post_create_script()
+                else:
+                    obj.reindexObject()
+                    notify(ObjectEditedEvent(obj))
+                    obj.at_post_edit_script()
+
+            # We'll set the layout after the validation and and even if there
+            # are no other changes.
+            if 'layout' in data:
+                layout = data['layout']
+                self.context.setLayout(layout)
+
+            # OrderingMixin
+            self.handle_ordering(data)
+
+            return obj
+
+        def validate(self):
+            # Instead of calling P.Archetypes.BaseObject.validate() we have to
+            # provide a custom validation implementation here because some
+            # validators extract the field value from the request. However a JSON
+            # API request does not contain any form values in the request.
+            # Thus we fake a request that extracts form values from the object on
+            # demand.
+
+            obj = self.context
+            request = ValidationRequest(self.request, obj)
+            errors = {}
+
+            obj.pre_validate(request, errors)
+
+            for pre_validator in subscribers((obj,), IObjectPreValidation):
+                pre_errors = pre_validator(request)
+                if pre_errors is not None:
+                    for field_name, error_message in pre_errors.items():
+                        if field_name in errors:
+                            errors[field_name] += " %s" % error_message
```

Comment:
shouldn't we use single quotes and `.format()` instead?

Reply:
If we don't enforce such a style decision we shouldn't require people to stick with it IMHO.